### PR TITLE
Remove the eip712 feature flag and have it enabled by default

### PR DESCRIPTION
**Pull request by dbelv**
## Motivation
Closes https://github.com/gakonst/ethers-rs/issues/2408

## Solution
Remove feature flag enabling by default.

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Breaking changes

### CODE REVIEW

**Comment by dbelv** [comment]
Muscle memory had me doing `cargo fmt` which is unfortunately why large amount of touched files within the same commit vs the actual implementation of the issue.

**Comment by dbelv** [comment]
Sorry I have overloaded the PR for clippy fixes as well

File: ethers-contract/Cargo.toml [hunk 1]
```diff
@@ -58,6 +58,3 @@ legacy = []
 
 rustls = ["ethers-contract-abigen/rustls"]
 openssl = ["ethers-contract-abigen/openssl"]
-
-# Deprecated
-eip712 = []
```
Comment (DaniPopes) [inline ethers-contract/Cargo.toml:63]: Could you add them back here too?

File: ethers-core/Cargo.toml [hunk 2]
```diff
@@ -70,6 +70,3 @@ rand.workspace = true
 celo = ["legacy"] # celo support extends the transaction format with extra fields
 legacy = []
 macros = ["syn", "cargo_metadata", "once_cell"]
-
-# Deprecated
-eip712 = []
```
Comment (DaniPopes) [inline ethers-core/Cargo.toml:75]: Same

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: ethers-core/README.md [hunk 3]
```diff
@@ -10,10 +10,6 @@ other utilities for interacting with the Ethereum ecosystem
 
 For more information, please refer to the [book](https://gakonst.com/ethers-rs).
 
-## Feature flags
-
--   `eip712`: Provides the `Eip712` trait and derive procedural macro for EIP-712 encoding of typed data.
```
Comment (prestwich) [inline ethers-core/README.md:15]: modify to say "does nothing"

File: ethers-core/src/types/signature.rs [hunk 4]
```diff
@@ -69,7 +69,6 @@ impl fmt::Display for Signature {
     }
 }
 
-#[cfg(feature = "eip712")]
```
Comment (prestwich) [inline ethers-core/src/types/signature.rs:72]: combine with main impl block

**Review by prestwich — CHANGES_REQUESTED** [review]
keep the flag in the Cargo.toml. Removing it will break compilation for library consumers

File: ethers-contract/Cargo.toml (hunk 1 above)
Comment (dbelv) [reply ethers-contract/Cargo.toml:63]: Ah right, yep.

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**